codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientProtocolHandler.java

```diff
@@ -200,12 +206,18 @@ public void handlerAdded(ChannelHandlerContext ctx) {
         if (cp.get(WebSocketClientProtocolHandshakeHandler.class) == null) {
             // Add the WebSocketClientProtocolHandshakeHandler before this one.
             ctx.pipeline().addBefore(ctx.name(), WebSocketClientProtocolHandshakeHandler.class.getName(),
-                    new WebSocketClientProtocolHandshakeHandler(handshaker));
+                                     new WebSocketClientProtocolHandshakeHandler(handshaker)
+                                             .handshakeTimeoutMillis(handshakeTimeoutMillis));
```

Comment:
Can we just pass this in via an additional constructor parameter? No need to introduce a mutable state.

Reply:
Done!